concent_api/core/tests/test_integration_get_task_result.py

```diff
@@ -0,0 +1,601 @@
+
+from core.tests.utils       import ConcentIntegrationTestCase
+from utils.testing_helpers  import generate_ecc_key_pair
+
+
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY    = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY     = CONCENT_PUBLIC_KEY,
+    CONCENT_MESSAGING_TIME = 10,  # seconds
+    FORCE_ACCEPTANCE_TIME  = 10,  # seconds
+)
+class GetTaskResultIntegrationTest(ConcentIntegrationTestCase):
+
+    def test_requestor_forces_get_task_result_and_concent_immediately_sends_rejection_due_to_exceeded_time_for_download(self):
+        """
+        Tests if on requestor ForceGetTaskResult message Concent will return ForceGetTaskResultRejected
+        if time for download is exceeded.
+
+        Expected message exchange:
+        Requestor -> Concent:    ForceGetTaskResult
+        Concent   -> Requestor:  ForceGetTaskResultRejected
+        """
+
+        # STEP 1: Requestor forces get task result via Concent.
+        # Concent rejects request immediately when download time is exceeded.
+        serialized_task_to_compute = self._get_serialized_task_to_compute(
+            timestamp   = "2017-12-01 10:00:00",
+            deadline    = "2017-12-01 11:00:00",
+        )
+        serialized_force_get_task_result = self._get_serialized_force_get_task_result(
+            task_to_compute = serialized_task_to_compute,
+            timestamp       = "2017-12-01 11:00:11",
+        )
+
+        with freeze_time("2017-12-01 11:00:11"):
+            response = self.client.post(
+                reverse('core:send'),
+                data                           = serialized_force_get_task_result,
+                content_type                   = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = self._get_encoded_requestor_public_key()
+            )
+
+        self.assertEqual(response.status_code,  200)
+
+        message_from_concent = load(response.content, self.REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        self.assertIsInstance(message_from_concent, message.ForceGetTaskResultRejected)
+        self.assertEqual(message_from_concent.timestamp, self._parse_iso_date_to_timestamp("2017-12-01 11:00:11"))
+        self.assertEqual(message_from_concent.reason, "ACCEPTANCE_TIME_LIMIT_EXCEEDED")
+
+    def test_requestor_forces_get_task_result_and_concent_immediately_sends_rejection_due_to_already_sent_message(self):
+        """
+        Tests if on requestor ForceGetTaskResult message Concent will return ForceGetTaskResultRejected
+        if ForceGetTaskResult was already sent by this requestor.
+
+        Expected message exchange:
+        Requestor -> Concent:    ForceGetTaskResult
+        Requestor -> Concent:    ForceGetTaskResult
+        Concent   -> Requestor:  ForceGetTaskResultRejected
+        """
+
+        # STEP 1: Requestor forces get task result via Concent.
+        # Response is not important here TODO: Is it ?
```

Comment:
Response indicates whether the submission was successful, so it is important. You should not get HTTP 4xx for example. You should test this response too.

Reply:
OK. So now I see more clearly what this test is about.

This is a perfect situation that shows you why we need `assert` in tests and why it's different from `self.assertEqual()` and other tests assertions.

You should check the result but only with `assert`. This first call is not part of the test it's part of your setup. You do the first call and then you assume that it succeeded. If it did not, there's no point in continuting the test. If it returns an error it's not the code that's broken. It's your test because you should have constructed it in such a way that this first call succeeds.

So, in short, you need to `assert` that the first submission is successful.

Reply:
:+1: